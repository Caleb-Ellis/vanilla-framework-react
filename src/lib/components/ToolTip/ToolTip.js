import React from 'react';
import PropTypes from 'prop-types';

const ToolTip = ({
  id, message, position, children,
}) => {
  const modifierClassPrefix = 'p-tooltip--';

  let className = 'p-tooltip';

  if (position) {
    className += ` ${modifierClassPrefix}${position}`;
  }

  return (
    <button className={className} aria-describedby={id}>
      {children}
      <span className="p-tooltip__message" role="tooltip" id={id}>
        {message}
      </span>
    </button>
  );
};

ToolTip.defaultProps = {
  position: '',
};

ToolTip.propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.string,
  id: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

ToolTip.displayName = 'ToolTip';

export default ToolTip;
