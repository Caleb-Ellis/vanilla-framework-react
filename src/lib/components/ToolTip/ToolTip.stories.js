import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import ToolTip from './ToolTip';

storiesOf('ToolTip', module)
  .add('Bottom left tooltip',
    withInfo('Use this tooltip when you want the message to appear at bottom left')(() => (
      <ToolTip id="default-tooltip" message="Lorem ipsum dolor sit amet">
        Bottom left tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Bottom center tooltip',
    withInfo('Use this tooltip when you want the message to appear at bottom center')(() => (
      <ToolTip id="btm-cntr" message="Lorem ipsum dolor sit amet" position="btm-center">
        Bottom center tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Bottom right tooltip',
    withInfo('Use this tooltip when you want the message to appear at bottom right')(() => (
      <ToolTip id="btm-right" message="Lorem ipsum dolor sit amet" position="btm-right">
        Bottom right tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Left tooltip',
    withInfo('Use this tooltip when you want the message to appear to the left')(() => (
      <ToolTip id="left" message="Lorem ipsum dolor sit amet" position="left">
        Left tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Right tooltip',
    withInfo('Use this tooltip when you want the message to appear to the right')(() => (
      <ToolTip id="right" message="Lorem ipsum dolor sit amet" position="right">
        Right tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Top left tooltip',
    withInfo('Use this tooltip when you want the message to appear at top left')(() => (
      <ToolTip id="to-lft" message="Lorem ipsum dolor sit amet" position="top-left">
        Top left tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Top center tooltip',
    withInfo('Use this tooltip when you want the message to appear at top center')(() => (
      <ToolTip id="tp-cntr" message="Lorem ipsum dolor sit amet" position="top-center">
        Top center tooltip
      </ToolTip>
    ),
    ),
  )
  .add('Top right tooltip',
    withInfo('Use this tooltip when you want the message to appear at top right')(() => (
      <ToolTip id="tp-right" message="Lorem ipsum dolor sit amet" position="top-right">
        Top right tooltip
      </ToolTip>
    ),
    ),
  );
