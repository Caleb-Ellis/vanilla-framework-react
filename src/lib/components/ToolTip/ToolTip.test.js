import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import ToolTip from './ToolTip';

describe('ToolTip component', () => {
  it('should render a tooltip component', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="default-tooltip" message="Lorem ipsum dolor sit amet">
        Bottom left tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a bottom center tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="btm-cntr" message="Lorem ipsum dolor sit amet" position="btm-center">
        Bottom center tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a bottomright tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="btm-right" message="Lorem ipsum dolor sit amet" position="btm-right">
        Bottom right tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a left tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="left" message="Lorem ipsum dolor sit amet" position="left">
        Left tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a right tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="right" message="Lorem ipsum dolor sit amet" position="right">
        Right tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a top left tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="to-lft" message="Lorem ipsum dolor sit amet" position="top-left">
        Top left tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a top center tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="tp-cntr" message="Lorem ipsum dolor sit amet" position="top-center">
        Top center tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a tooltip component with a top right tooltip', () => {
    const toolTip = ReactTestRenderer.create(
      <ToolTip id="tp-right" message="Lorem ipsum dolor sit amet" position="top-right">
        Top right tooltip
      </ToolTip>,
    );
    const json = toolTip.toJSON();
    expect(json).toMatchSnapshot();
  })
});
