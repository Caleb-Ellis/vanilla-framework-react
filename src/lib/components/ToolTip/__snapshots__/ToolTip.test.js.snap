// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolTip component should render a tooltip component 1`] = `
<button
  aria-describedby="default-tooltip"
  className="p-tooltip"
>
  Bottom left tooltip
  <span
    className="p-tooltip__message"
    id="default-tooltip"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a bottom center tooltip 1`] = `
<button
  aria-describedby="btm-cntr"
  className="p-tooltip p-tooltip--btm-center"
>
  Bottom center tooltip
  <span
    className="p-tooltip__message"
    id="btm-cntr"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a bottomright tooltip 1`] = `
<button
  aria-describedby="btm-right"
  className="p-tooltip p-tooltip--btm-right"
>
  Bottom right tooltip
  <span
    className="p-tooltip__message"
    id="btm-right"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a left tooltip 1`] = `
<button
  aria-describedby="left"
  className="p-tooltip p-tooltip--left"
>
  Left tooltip
  <span
    className="p-tooltip__message"
    id="left"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a right tooltip 1`] = `
<button
  aria-describedby="right"
  className="p-tooltip p-tooltip--right"
>
  Right tooltip
  <span
    className="p-tooltip__message"
    id="right"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a top center tooltip 1`] = `
<button
  aria-describedby="tp-cntr"
  className="p-tooltip p-tooltip--top-center"
>
  Top center tooltip
  <span
    className="p-tooltip__message"
    id="tp-cntr"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a top left tooltip 1`] = `
<button
  aria-describedby="to-lft"
  className="p-tooltip p-tooltip--top-left"
>
  Top left tooltip
  <span
    className="p-tooltip__message"
    id="to-lft"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;

exports[`ToolTip component should render a tooltip component with a top right tooltip 1`] = `
<button
  aria-describedby="tp-right"
  className="p-tooltip p-tooltip--top-right"
>
  Top right tooltip
  <span
    className="p-tooltip__message"
    id="tp-right"
    role="tooltip"
  >
    Lorem ipsum dolor sit amet
  </span>
</button>
`;
